geopandas/io/file.py

```diff
@@ -541,6 +541,9 @@ def _to_file(
             stacklevel=3,
         )
 
+    if (df.dtypes == "geometry").sum() > 1:
+        raise ValueError("GeoDataFrame has to have single indices")
```

Comment:
I am not sure if this error message actually captures the issue. I would suggest something along these lines

```
GeoDataFrame contains multiple geometry columns but the selected driver supports only a single geometry column. Use a different driver, drop additional geometry columns or convert them to a supported format like well-known text (WKT) using `GeoSeries.to_wkt()`.
```